CRM/Core/Page/API/3/Rest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * Class to handle requests for APIv3 Rest Interface.
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+class CRM_Core_Page_API_3_Rest extends CRM_Core_Page {
+
+  public function run() {
+    if (defined('PANTHEON_ENVIRONMENT')) {
+      ini_set('session.save_handler', 'files');
+    }
+    $rest = new CRM_Utils_REST();
+
+    // Json-appropriate header will be set by CRM_Utils_Rest
+    // But we need to set header here for non-json
+    if (empty($_GET['json'])) {
+      header('Content-Type: text/xml');
+    }
+    echo $rest->bootAndRun();
```

Comment:
@seamuslee001 I haven't tested yet but just a quick observation/question, this calls `loadCMSBootstrap()`, wouldn't that attempt to bootstrap the CMS again?

ping @kcristiano @christianwach

Reply:
@mecachisenros it will and that is what the `extern/rest.php` did.  Called from extern I understand we had to boot the CMS.  The question here is do we have to?  what would we need the CMS for?  

@seamuslee001 do you think booting the CMS is needed?

Reply:
Couldn't there be a short-circuit in [`loadBootStrap()`](https://github.com/civicrm/civicrm-core/blob/f71afe4a955d2848ccad348a1c74ee2ebf49b404/CRM/Utils/System/WordPress.php#L577) which bails early if, say, `ABSPATH` is defined?

Reply:
@kcristiano @mecachisenros @christianwach 
So remember we are talking about the rest API here we are with this essentially exposing a public route for people to pass the rest API credentials and we would need to run the boot and run because we need to take the API_Key and site_key from the request and boot the CMS with the permissions of the user that API_Key is linked to right so that whatever API action is to be taken is to be taken as per their permissions. So we need to re-boot the CMS to effectively login the user

Reply:
@seamuslee001 But... doesn't that route already pass through WordPress, in which case it's already bootstrapped?

Reply:
Actually yes @mecachisenros @christianwach @kcristiano I think we can avoid the re-bootstrapping pushing an update now